### PR TITLE
Describe troubleshooting techniques for Streams in SCDF

### PR DESCRIPTION
**Pull request by chrisjs**
Resolves https://github.com/spring-cloud/spring-cloud-dataflow/issues/3150

### CODE REVIEW

File: content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md [hunk 1]
```diff
@@ -4,7 +4,134 @@ title: 'Debugging Stream applications deployed by Data Flow'
...
+    A --> |No| C{Exceptions in app logs?}
+	C --> |Yes| D(Resolve exceptions)
+```
+
+By default, the amount of memory allocated to Docker may be too low.
+The command `docker stats` can provide useful information into resource usage.
+If applications are failing to deploy due to resource constraints, increase resource allocations.
+Consult the [Docker documentation](https://docs.docker.com/) for your platform.
+
+As streams are deployed via Skipper, applications that are part of that stream will be launched as Java processes on the Skipper container.
```
Comment (sabbyanandan) [inline content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md]: Perhaps also add a note about "the more the number of streams, the more memory is required for skipper container to run the apps in them" or something like that. 

See https://github.com/spring-cloud/spring-cloud-dataflow/issues/3107#issuecomment-477250327 for background.

File: content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md [hunk 2]
```diff
@@ -4,7 +4,134 @@ title: 'Debugging Stream applications deployed by Data Flow'
...
+
+```mermaid
+graph TD;
+    A{Errors in pod events table?} --> |Yes| B(Resolve)
+    C{Exceptions in app pod logs?} --> |Yes| B
+```
+
+When describing a pod, the `events` table section provides useful information when debugging and can be invoked by the following:
+
+`kubectl describe po/pod_name`
```
Comment (sabbyanandan) [inline content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md]: Perhaps paste some sample output for reference?

**Comment by sabbyanandan** [comment]
Maybe also towards the end, we can drop a statement like this:

> If none of those above troubleshooting techniques helped and if you're still looking for help, you can reach out to us in [StackOverflow](https://stackoverflow.com/tags/spring-cloud-dataflow/) with the relevant details (see: [Wiki](https://github.com/spring-cloud/spring-cloud-dataflow/wiki/Reporting-Issues)) - we actively monitor the forum threads.

File: content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md [hunk 3]
```diff
@@ -4,7 +4,134 @@ title: 'Debugging Stream applications deployed by Data Flow'
...
+# Docker Compose - Startup
+
+```mermaid
+graph TD;
+    A{DATAFLOW_VERSION defined?} --> |No| A
+    A --> |Yes| B{SKIPPER_VERSION defined?}
+	B --> |No| B
+```
+
+The environment variables `DATAFLOW_VERSION` and `SKIPPER_VERSION` must be available in the current shell environment via `export` or prefixing the `docker-compose` command.
```
[the remark this answers is not in the document]
Comment (chrisjs) [reply content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md]: updated

File: content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md [hunk 4]
```diff
@@ -4,7 +4,134 @@ title: 'Debugging Stream applications deployed by Data Flow'
...
+# Docker Compose - Runtime
+
+```mermaid
+graph TD;
+    A{Low Resources?} --> |Yes| B(Increase Docker Resources)
+    A --> |No| C{Exceptions in app logs?}
+	C --> |Yes| D(Resolve exceptions)
+```
+
+By default, the amount of memory allocated to Docker may be too low.
```
[the remark this answers is not in the document]
Comment (chrisjs) [reply content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md]: added 8GB as the min

File: content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md (hunk 1 above)
Comment (chrisjs) [reply content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md]: updated

File: content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md (hunk 2 above)
Comment (chrisjs) [reply content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md]: i added a "successful" events table from deploying a `time` application. figured adding something with errors might be confusing and require additional explination

**Comment by chrisjs** [comment]
> Maybe also towards the end, we can drop a statement like this:
> 
> > If none of those above troubleshooting techniques helped and if you're still looking for help, you can reach out to us in [StackOverflow](https://stackoverflow.com/tags/spring-cloud-dataflow/) with the relevant details (see: [Wiki](https://github.com/spring-cloud/spring-cloud-dataflow/wiki/Reporting-Issues)) - we actively monitor the forum threads.

added new section with this

**Comment by chrisjs** [comment]
merged in 284ba70a4652f03ad3150e0d131ac48aa87fe98c

**Pull request closed without merging**